### PR TITLE
feat: departments

### PR DESCRIPTION
**Pull request by KambojRajan**
=> Departments page initiated, enhancing website functionality.
=> Added "About", "HOD's Message", and "Mission and Vision" sections.

### CODE REVIEW

File: tsconfig.json
Comment (GetPsyched) [inline tsconfig.json:1]: Please run `npm run format` to avoid pushing diffs like these

**Review by GetPsyched — CHANGES_REQUESTED** [review]
This is too segregated. Reduce the number of components.
cc @maan2003

**Comment by maan2003** [comment]
why am i cced?

File: pnpm-lock.yaml
Comment (maan2003) [inline pnpm-lock.yaml:1]: pnpm?

File: components/DepartmentRelated/maxWidthWrapper.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    <div
+      className={clsx(
+        'mx-auto w-full max-w-screen-xl px-2.5 md:px-20',
+        className
+      )}
+    >
+      {children}
+    </div>
+  );
+};
```
Comment (maan2003) [inline components/DepartmentRelated/maxWidthWrapper.tsx:21]:
```suggestion
const maxWidthStyle = 'mx-auto w-full max-w-screen-xl px-2.5 md:px-20';
```

Comment (maan2003) [reply components/DepartmentRelated/maxWidthWrapper.tsx:21]: likely some similar in mechanism in tailwind as well

File: components/DepartmentRelated/subNav.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    { key: 'FacultyAndMore', title: 'Faculty And More', link: '#' },
+    { key: 'PhotoGallery', title: 'Photo Gallery', link: '#' },
+  ];
+
+  return (
+    <div className="mb-3 backdrop-blur-md">
+      <div className="border-1 flex justify-evenly space-x-10 rounded-full border  border-white bg-opacity-30 p-3 px-4">
+        {subLinks.map((btn) => (
+          <Link
+            key={btn.key}
```
Comment (maan2003) [inline components/DepartmentRelated/subNav.tsx:18]: any use of key in a static list?

**Comment by GetPsyched** [comment]
> why am i cced?

Since you commented on #59 regarding components, thought I'd confirm my opinion here.

File: components/DepartmentRelated/subNav.tsx (hunk 2 above)
Comment (GetPsyched) [reply components/DepartmentRelated/subNav.tsx:18]: React shoots warnings if the `key` attribute is not set in places where you're mapping an array, even if it's static.

**Comment by maan2003** [comment]
I asked for components in #59 because it was repeating a lot of code.

I don't understand why components are split here? because one file was getting too big?

**Comment by GetPsyched** [comment]
> I don't understand why components are split here? because one file was getting too big?

Same question. I don't think it's justified.

Anyway, feel free to unsubscribe.

**Review by GetPsyched — CHANGES_REQUESTED** [review]
Rebase and resolve conflicts please

File: package.json [hunk 3]
```diff
@@ -18,14 +18,20 @@
   "dependencies": {
     "@formatjs/intl-localematcher": "^0.5.4",
     "@prisma/client": "^5.9.1",
+    "@radix-ui/react-icons": "^1.3.0",
```
Comment (GetPsyched) [inline package.json:21]: Use native react-icons

File: styles/globals.css [hunk 4]
```diff
@@ -1,10 +1,18 @@
-@import url('https://fonts.googleapis.com/css2?family=DM+Serif+Display:ital@0;1&display=swap');
-@import url('https://fonts.googleapis.com/css2?family=DM+Sans:ital,opsz,wght@0,9..40,100..1000;1,9..40,100..1000&family=DM+Serif+Display:ital@0;1&display=swap');
+/* @import url('https://fonts.googleapis.com/css2?family=DM+Serif+Display:ital@0;1&display=swap');
```
Comment (GetPsyched) [inline styles/globals.css:1]: Don't comment this

**Review by GetPsyched — CHANGES_REQUESTED** [review]
Make all file names kebab-case

File: .env.example [hunk 5]
```diff
@@ -4,4 +4,4 @@ GOOGLE_CLIENT_SECRET=""
 
 # next-auth configuration
 NEXTAUTH_URL="http://localhost:3000"
-NEXTAUTH_SECRET=""
+NEXTAUTH_SECRET=""
```
Comment (GetPsyched) [inline .env.example:7]: I think you removed the newline at the end here. There shouldn't be a diff here.

File: app/[locale]/directors-corner.tsx [hunk 6]
```diff
@@ -53,4 +53,4 @@ export default async function DirectorsCorner({ locale }: { locale: string }) {
       </section>
     </article>
   );
-}
+}
```
Comment (GetPsyched) [inline app/[locale]/directors-corner.tsx:56]: Weird, you don't have new lines at the end of any file. Resolve this.

File: styles/globals.css [hunk 7]
```diff
@@ -5,6 +5,14 @@
 @tailwind components;
 @tailwind utilities;
 
+.remove-arrow::-webkit-inner-spin-button,
+.remove-arrow::-webkit-outer-spin-button {
+  -webkit-appearance: none;
+  margin: 0;
+}
+.remove-arrow {
```
Comment (GetPsyched) [inline styles/globals.css:13]: Line breaks not added here either

**Review by GetPsyched — CHANGES_REQUESTED** [review]

File: .env.example (hunk 5 above)
Comment (GetPsyched) [reply .env.example:7]: I don't think this is resolved? At least I don't see changes on my phone

File: app/[locale]/notifications.tsx
[the remark this answers is not in the document]
Comment (GetPsyched) [reply app/[locale]/notifications.tsx:1]: Don't mark as resolved if not resolved.

File: lib/utils.ts
Comment (GetPsyched) [inline lib/utils.ts:1]: no diff here

File: app/[locale]/profile/page.tsx
Comment (GetPsyched) [inline app/[locale]/profile/page.tsx:1]: rebase on master

**Review by GetPsyched — CHANGES_REQUESTED** [review]
Needs rebase.

Do NOT merge master into this branch. Do a rebase.

File: components/ui/button.tsx [hunk 8]
```diff
@@ -1,8 +1,7 @@
 import * as React from 'react';
 import { Slot } from '@radix-ui/react-slot';
 import { cva, type VariantProps } from 'class-variance-authority';
-
-import { cn } from '~/lib/utils';
```
Comment (GetPsyched) [inline components/ui/button.tsx]: Why? Feel free to use `cn`

File: components/ui/card.tsx
Comment (GetPsyched) [inline components/ui/card.tsx:1]: Same here

**Review by GetPsyched — CHANGES_REQUESTED** [review]

File: components.json
Comment (GetPsyched) [inline components.json:1]: hm?

**Comment by KambojRajan** [comment]
image

Could you please provide a hint on how to achieve this? I have sought help from the internet, but I am unable to achieve this effect. and this is the only thing that's left

**Comment by KambojRajan** [comment]
if is so, then everything is done

File: i18n/translations.ts [hunk 9]
```diff
@@ -39,4 +39,10 @@ export interface Translations {
   NotFound: { title: string; description: string; backHome: string };
   Unauthorised: { title: string; description: string };
   WorkInProgress: { title: string; description: string };
+  Departments: {
+    subLinks: { key: string; title: string; link: string }[];
+    departmentTitles: string[];
+    sectionTitles: string[];
+    readMore: string;
+  };
```
Comment (GetPsyched) [inline i18n/translations.ts:47]: Please put this alphabetically.

File: i18n/hi.ts
Comment (GetPsyched) [inline i18n/hi.ts:1]: Here as well

File: i18n/en.ts
Comment (GetPsyched) [inline i18n/en.ts:1]: And here

**Review by GetPsyched — CHANGES_REQUESTED** [review]

File: app/[locale]/notifications.tsx
Comment (GetPsyched) [inline app/[locale]/notifications.tsx:1]: As mentioned before, this diff should be removed.

File: i18n/translations.ts [hunk 10]
```diff
@@ -39,4 +39,10 @@ export interface Translations {
   NotFound: { title: string; description: string; backHome: string };
   Unauthorised: { title: string; description: string };
   WorkInProgress: { title: string; description: string };
+  Departments: {
+    departmentTitles: string[];
+    readMore: string;
+    sectionTitles: string[];
+    subLinks: { key: string; link: string; title: string }[];
+  };
```
Comment (GetPsyched) [inline i18n/translations.ts:47]: I meant that this is not alphabetical with respect to the other elements like `WorkInProgress`. The inside stuff can be in any order.

File: i18n/translations.ts [hunk 11]
```diff
@@ -3,6 +3,12 @@ export async function getTranslations(locale: string): Promise<Translations> {
 }
 
 export interface Translations {
+  Departments: {
+    departmentTitles: string[];
+    readMore: string;
+    sectionTitles: string[];
+    subLinks: { key: string; link: string; title: string }[];
```
Comment (GetPsyched) [inline i18n/translations.ts:10]: Huh, the `key` and `link` don't need translations, remove it from here.

File: i18n/en.ts [hunk 12]
```diff
@@ -1,6 +1,43 @@
 import type { Translations } from './translations';
 
 const text: Translations = {
+  Departments: {
+    subLinks: [
```
Comment (GetPsyched) [inline i18n/en.ts:5]: Sorted or not, this is now in a different order than in the other translation files.

File: app/api/route.ts [hunk 13]
```diff
@@ -1,4 +1,7 @@
 export async function GET(request: Request) {
+  if (request.method !== 'GET') {
+    console.log('No other method required');
+  }
```
Comment (GetPsyched) [inline app/api/route.ts:4]: Uh, what?

1. Why are you editing this API route?
2. A function named `GET` here will always catch `GET` requests, so this check is meaningless; see <https://nextjs.org/docs/app/api-reference/file-conventions/route#http-methods>

File: app/[locale]/academics/departments/page.tsx [hunk 14]
```diff
@@ -1,9 +1,70 @@
...
+];
+
+export default async function Departments({
   params: { locale },
 }: {
   params: { locale: string };
 }) {
-  return <WorkInProgress locale={locale} />;
+  const text = (await getTranslations(locale)).Departments;
+  return (
```
Comment (GetPsyched) [inline app/[locale]/academics/departments/page.tsx:45]:
```suggestion
  const text = (await getTranslations(locale)).Departments;

  return (
```

File: app/[locale]/academics/departments/[name]/page.tsx [hunk 15]
```diff
@@ -1,9 +1,239 @@
...
+
+  return (
+    <main className=" w-screen">
+      <header
+        style={{
+          backgroundImage: `url(${departmentData.titleImage})`,
+          backgroundSize: 'cover',
+          backgroundPosition: 'center',
+        }}
+      >
```
Comment (GetPsyched) [inline app/[locale]/academics/departments/[name]/page.tsx:43]: Why do we have inline styles over Tailwind? Please fix. We can set bg images using Tailwind as well, see <https://tailwindcss.com/docs/background-image#arbitrary-values>

**Review by GetPsyched — CHANGES_REQUESTED** [review]

File: app/api/route.ts (hunk 13 above)
Comment (KambojRajan) [reply app/api/route.ts:4]: sir, I encountered lint errors indicating that the request cannot be left unused. While I know it is redundant, I chose to include it just for the sake of the error, and I'm aware of accessing documentation independently

File: app/[locale]/academics/departments/[name]/page.tsx (hunk 15 above)
Comment (KambojRajan) [reply app/[locale]/academics/departments/[name]/page.tsx:43]: Thank you for informing me. I wasn't aware of this, but now that I know, I'll make the necessary changes

File: app/api/route.ts (hunk 13 above)
Comment (GetPsyched) [reply app/api/route.ts:4]: If it's redundant you can just remove the parameter from the function, or rename it to `_request`.

Anyways, this change is not in the scope of this PR, remove it

Comment (KambojRajan) [reply app/api/route.ts:4]: ok, I will remove it

File: i18n/en.ts (hunk 12 above)
Comment (KambojRajan) [reply i18n/en.ts:5]: Sir, may I suggest placing the department at the 0th index and then proceeding to copy and paste whatever content is present in the master? If this is not the requirement, please tell me.

File: app/api/route.ts [hunk 16]
```diff
@@ -1,4 +1,4 @@
-export async function GET(request: Request) {
+export async function GET(_request: Request) {
```
Comment (GetPsyched) [inline app/api/route.ts:1]: Huh?

Don't change anything here please. As I said before:
> Anyways, this change is not in the scope of this PR, remove it

**Review by GetPsyched — CHANGES_REQUESTED** [review]

File: i18n/en.ts (hunk 12 above)
Comment (GetPsyched) [reply i18n/en.ts]: I'm confused. What are you trying to say? Copy paste what's in master? Wdym?

File: app/[locale]/academics/departments/[name]/page.tsx [hunk 17]
```diff
@@ -34,21 +34,21 @@ export default async function Department({
...
-          backgroundPosition: 'center',
-        }}
-      >
+      <header className="relative h-[481px] w-full">
+        <Image
+          alt="department image"
+          className="-z-50 object-cover"
+          src={departmentData.titleImage}
+          fill
+        />
```
[the remark this answers is not in the document]
Comment (KambojRajan) [reply app/[locale]/academics/departments/[name]/page.tsx:43]: Sir hame jitna pata hai hum dynamic links nahi de sakte han bg-[url(...)] me, toh agar apko koi tarika pata hai jo bina inline style k hota hai toh bata dijiye

File: app/[locale]/academics/departments/[name]/page.tsx [hunk 18]
```diff
@@ -202,14 +202,7 @@ export default async function Department({
...
-            backgroundSize: 'cover',
-            opacity: 0.7,
-          }}
-        >
+        <article className="relative mt-20 flex flex-1 justify-between space-x-4 bg-[url('https://s3-alpha-sig.figma.com/img/32a3/273b/70051ff8670610a0e06d9ff49e7cb675?Expires=1710115200&Key-Pair-Id=APKAQ4GOSFWCVNEHN3O4&Signature=V6vPstFQBLAj0l7cTLalslHY9fulCefKOvsOineaOXTkm0l4aXO14Q4XlTZP7IsU~Snm3gIr2ITRgpLjRXcoGeokV~I0dV6VcdUeZyWWIEtJa9y0yzn6mqXpGK47f9jKorWWjYgowENmLteeS93qLuSa9ozxzFRqaSunXNfTHj76ltbLC1xtVumVpLVbf~ywkh~lDsyA-VyspcY3gVYnjGi6NaRHcLtJHZVgbIxxGEoG8rzSXzc2OmjRAEgoa8HeGI-WllViUXrDpqIlBu1HLac~a4kV6yV0vVDCwaHZ2-I9ySAiHM-4wWySA77o1acMODptO1KT7LirkEjSUYWASA__')] bg-cover bg-clip-content p-10 opacity-55">
```
[the remark this answers is not in the document]
Comment (KambojRajan) [reply app/[locale]/academics/departments/[name]/page.tsx:205]: because the url was static that's why

File: app/api/route.ts (hunk 16 above)
Comment (KambojRajan) [reply app/api/route.ts:1]: ok, I wont

File: app/[locale]/academics/departments/[name]/page.tsx (hunk 17 above)
[the remark this answers is not in the document]
Comment (GetPsyched) [reply app/[locale]/academics/departments/[name]/page.tsx:43]:
```ts
className=`... bg-[url(${titleImage})] ...`
```

[the remark this answers is not in the document]
Comment (KambojRajan) [reply app/[locale]/academics/departments/[name]/page.tsx:43]: No sir, I had tried that way, and nextjs tries to import it as a module 
image
I check the issue on stack overflow there it is suggested to use style

[the remark this answers is not in the document]
Comment (GetPsyched) [reply app/[locale]/academics/departments/[name]/page.tsx:43]: Hm, weird. Let it be as it is then; I'll fix it later.

**Comment by GetPsyched** [comment]
Closed in favour of #111

**Pull request closed without merging**